centreon/www/front_src/src/Header/Resources/Host/getHostPropsAdapter.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  labelDownStatusHosts,
+  labelUnreachableStatusHosts,
+  labelUpStatusHosts,
+  labelAll,
+  labelDown,
+  labelPending,
+  labelUnreachable,
+  labelUp
+} from './translatedLabels';
+
+type ChangeFilterAndNavigate = (params: {
+  criterias: Array<Criteria>;
+  link: string;
+}) => (e: React.MouseEvent<HTMLLinkElement>) => void;
+
+export interface HostPropsAdapterOutput {
+  counters: CounterProps['counters'];
+  hasPending: boolean;
+  items: SubMenuProps['items'];
+}
+
+type GetHostPropsAdapter = Adapter<HostStatusResponse, HostPropsAdapterOutput>;
+
+const formatCount = (
+  unhandled: number | string,
+  total: number | string
+): string =>
+  `${numeral(unhandled).format('0a')}/${numeral(total).format('0a')}`;
+
+const getHostPropsAdapter: GetHostPropsAdapter = ({
+  useDeprecatedPages,
+  applyFilter,
+  navigate,
+  t,
+  data
+}) => {
+  const changeFilterAndNavigate: ChangeFilterAndNavigate =
+    ({ link, criterias }) =>
+    (e) => {
+      e.preventDefault();
+      if (!useDeprecatedPages) {
+        applyFilter({ criterias, id: '', name: 'New Filter' });
+      }
+
+      navigate(link);
+    };
+
+  const unhandledDownHostsCriterias = getDefaultCriterias({
+    resourceTypes: hostCriterias.value,
+    states: unhandledStateCriterias.value,
+    statuses: downCriterias.value as Array<SelectEntry>
+  });
+  const unhandledDownHostsLink = useDeprecatedPages
+    ? '/main.php?p=20202&o=h_down&search='
+    : getHostResourcesUrl({
+        stateCriterias: unhandledStateCriterias,
+        statusCriterias: downCriterias
+      });
+
+  const unhandledUnreachableHostsCriterias = getDefaultCriterias({
+    resourceTypes: hostCriterias.value,
+    states: unhandledStateCriterias.value,
+    statuses: unreachableCriterias.value as Array<SelectEntry>
+  });
+  const unhandledUnreachableHostsLink = useDeprecatedPages
+    ? '/main.php?p=20202&o=h_unreachable&search='
+    : getHostResourcesUrl({
+        stateCriterias: unhandledStateCriterias,
+        statusCriterias: unreachableCriterias
+      });
+
+  const upHostsCriterias = getDefaultCriterias({
+    resourceTypes: hostCriterias.value,
+    statuses: upCriterias.value as Array<SelectEntry>
+  });
+  const upHostsLink = useDeprecatedPages
+    ? '/main.php?p=20202&o=h_up&search='
+    : getHostResourcesUrl({
+        statusCriterias: upCriterias
+      });
+
+  const hostsCriterias = getDefaultCriterias({
+    resourceTypes: hostCriterias.value
+  });
+  const hostsLink = useDeprecatedPages
+    ? '/main.php?p=20202&o=h&search='
+    : getHostResourcesUrl();
+
+  const pendingHostsCriterias = getDefaultCriterias({
+    resourceTypes: hostCriterias.value,
+    statuses: pendingCriterias.value as Array<SelectEntry>
+  });
+  const pendingHostsLink = useDeprecatedPages
+    ? '/main.php?p=20202&o=h_pending&search='
+    : getHostResourcesUrl({
+        statusCriterias: pendingCriterias
+      });
+
+  const config = {
+    all: {
+      count: numeral(data.total).format('0a'),
```

Comment:
As you are using the same format on each counters. You can create an util function that wrap this format in order to avoid code duplication